test/Quaternion.jl

```diff
@@ -571,4 +579,54 @@ end
             @test_throws DivideError lyap(null, null)
         end
     end
+
+    @testset "widen" begin
+        @test widen(Quaternion{Int}) === Quaternion{Int128}
+        @test widen(QuaternionF32) === QuaternionF64
+        @test widen(QuaternionF64) === Quaternion{BigFloat}
+        @test widen(quat(1, 2, 3, 4)) === Quaternion{Int128}(1, 2, 3, 4)
+        q = rand(QuaternionF32)
+        @test widen(q) == convert(QuaternionF64, q)
+        q = rand(QuaternionF64)
+        @test widen(q) == convert(Quaternion{BigFloat}, q)
+    end
+
+    @testset "flipsign" begin
+        q = rand(QuaternionF64)
+        @test flipsign(q, 2) == q
+        @test flipsign(q, -3) == -q
+    end
+
+    @testset "read/write" begin
+        @testset "$T" for T in (Int16, Float32, Float64)
+            io = IOBuffer(; read=true, write=true)
+            q = rand(Quaternion{T})
+            write(io, q)
+            seek(io, 0)
+            q2 = read(io, Quaternion{T})
+            @test q == q2
+        end
+    end
+
+    @testset "big" begin
+        @test big(Quaternion{Int}) === Quaternion{BigInt}
+        @test big(QuaternionF64) === Quaternion{BigFloat}
+        @test big(quat(1, 2, 3, 4)) == Quaternion{BigInt}(1, 2, 3, 4)
+        q = rand(QuaternionF64)
+        @test big(q) == convert(Quaternion{BigFloat}, q)
+    end
+
+    @testset "round" begin
+        q = quat(1.1, 2.5, -3.5, 2.3)
+        @test round(q) == quat(1.0, 2.0, -4.0, 2.0)
+        @test round(q; digits=1) == q
+        @test round(q, RoundUp) == quat(2.0, 3.0, -3.0, 3.0)
+        @test round(q, RoundUp; digits=1) == q
+        @test round(q, RoundUp, RoundToZero) == quat(2.0, 2.0, -3.0, 2.0)
+        @test round(q, RoundUp, RoundToZero; digits=1) == q
+        @test round(q, RoundUp, RoundDown, RoundNearestTiesAway, RoundToZero) ==
+            quat(2.0, 2.0, -4.0, 2.0)
+        @test round(q, RoundUp, RoundDown, RoundNearestTiesAway, RoundToZero; digits=1) ==
+            q
```

Comment:
I thought these line breaks can be dropped. If there is a limit to the number of characters per line, it is better to add `.JuliaFormatter.toml` file. (Adding JuliaFormatter can be separated into another PR.)